File: ruby/elements/title.rb
require_relative 'image'

module Elements
  class Title < Element
    attr_reader :text

    def initialize( compiler, article, lines)
      @text = lines[0]
      if SPECIAL_CHARS =~ @text
        article.error( 'Bad characters in title')
        @discard = true
      end
    end

    def multiline?
      false
    end

    def page_content?
      false
    end

    def special?
      true
    end
  end
end

File: java/src/com/alofmethbin/jadawin/elements/Title.java
package com.alofmethbin.jadawin.elements;

import com.alofmethbin.jadawin.Article;
import java.util.List;

public class Title extends Element {
    private String text;
    
    public Title( Article article, List<String> lines) {
        super( article);
        text = checkSpecialChars( lines.get(0));
    }

    @Override
    public boolean isMultiline() {
        return false;
    }

    public String text() {
        return text;
    }
}
